### PR TITLE
refactor(limits): Concise implementation for workers/global/limits.ts

### PR DESCRIPTION
**Pull request by zharinov**
Ref: #7076, #7136

### CODE REVIEW

File: lib/workers/global/limits.spec.ts [hunk 1]
```diff
@@ -1,21 +1,43 @@
...
+
+  it('defaults to unlimited', () => {
+    expect(isLimitReached(Limit.Commits)).toBe(false);
+  });
+
+  it('increments undefined', () => {
+    incLimitedValue(Limit.Commits);
+    expect(isLimitReached(Limit.Commits)).toBe(false);
+    setMaxLimit(Limit.Commits, 1);
+    expect(isLimitReached(Limit.Commits)).toBe(true);
```
Comment (zharinov) [inline lib/workers/global/limits.spec.ts:41]: Not necessary the case, but is possible. Not sure, maybe it's better to reset `current` counter to `0` instead.

File: lib/util/git/index.ts [hunk 2]
```diff
@@ -18,6 +18,7 @@ import { logger } from '../../logger';
 import { ExternalHostError } from '../../types/errors/external-host-error';
 import { GitOptions, GitProtocol } from '../../types/git';
 import * as limits from '../../workers/global/limits';
+import { Limit } from '../../workers/global/limits';
```
Comment (viceice) [inline lib/util/git/index.ts:21]: Can you merge the imports please

File: lib/workers/global/index.ts [hunk 3]
```diff
@@ -9,6 +9,7 @@ import * as repositoryWorker from '../repository';
 import { autodiscoverRepositories } from './autodiscover';
 import { globalFinalize, globalInitialize } from './initialize';
 import * as limits from './limits';
+import { Limit } from './limits';
```
Comment (viceice) [inline lib/workers/global/index.ts:12]: Here too. More below. 🙃

File: lib/workers/global/initialize.ts [hunk 4]
```diff
@@ -35,7 +36,7 @@ export async function globalInitialize(
   config = await initPlatform(config);
   config = await setDirectories(config);
   packageCache.init(config);
-  limits.init(config);
+  limits.setMaxLimit(Limit.Commits, config.prCommitsPerRunLimit);
```
Comment (viceice) [inline lib/workers/global/initialize.ts:39]: Why not have a `init` in `limits` which calls `setMaxLimit` for each type? Do we have multiple init places?

Comment (zharinov) [reply lib/workers/global/initialize.ts:39]: Right, here we init `Commits` limit from config, but later we will use more dynamic variant for `PullRequests` and `Branches`

**Comment by zharinov** [comment]
Ready for review/merge

**Review by viceice — APPROVED** [review]

**Comment by rarkins** [comment]
Important that this is well tested in practice for existing use cases

**Comment by zharinov** [comment]
Can confirm it works with `--pr-commits-per-run-limit=X`

File: lib/workers/global/limits.spec.ts (hunk 1 above)
Comment (zharinov) [reply lib/workers/global/limits.spec.ts:41]: We're now resetting counter after `setMaxLimit()` call, since it's more intuitive behavior

**Pull request merged**